components/AppNav.tsx

```diff
@@ -42,23 +44,31 @@ const AuthLink = () => (
   </div>
 )
 
-const AuthButton: React.FC<AuthButtonProps> = ({ initial, username }) => (
-  <div>
-    <Button
-      btnType="border btn-secondary overflow-hidden p-2 text-truncate"
-      initial={initial}
-      text={username}
-    />
-    <Button
-      text="Logout"
-      btnType="border btn-secondary ml-2"
-      onClick={async () => {
-        const res = await logoutUser()
-        if (res) window.location.pathname = '/'
-      }}
-    />
-  </div>
-)
+const AuthButton: React.FC<AuthButtonProps> = ({ initial, username }) => {
+  const [logoutUser, { data }] = useMutation(LOGOUT_USER)
+  useEffect(() => {
+    const { success } = _.get(data, 'logout', false)
+    if (success) {
+      window.location.pathname = '/'
```

Comment:
`router.push('/')`

```
import { useRouter } from 'next/router'
...
const router = useRouter()
...

```

Reply:
We're no longer going to refactor to router.push since router doesn't trigger reload. There isn't a nice way for router to reload the page.